tests/accelerator/cuda/common/test_cuda_basics.cuh

```diff
@@ -26,11 +24,63 @@ class test_cuda_basics : public test_basics_base<T> {
 
  public:
   /// Constructor, setting up the inputs for all of the tests
-  test_cuda_basics() : test_basics_base<T>(m_resource) {}
+  test_cuda_basics() : test_basics_base<T>() {}
 
  protected:
-  /// Memory resource for all of the tests.
-  vecmem::cuda::managed_memory_resource m_resource;
+  /// Function setting things up for a test
+  virtual void SetUp() override {
+
+    // Create the vectors.
+    CUDA_ERROR_CHECK(cudaMallocManaged(
```

Comment:
I am **not** on board with this.

I agree with getting rid of the "vecmem frontend" of the code. But we still shall continue to use vecmem to test the code. Since we're using vecmem for the memory management of everything in all the downstream projects as well. So we should test the synergy between these codebases already here.

I.e. the project will definitely need to continue building vecmem for its tests.

Note that once we merge the project into detray, this is not "anything" even. Since detray is not going to stop depending on vecmem either.